src/features/vote/RevealPhase.tsx

```diff
@@ -0,0 +1,290 @@
+                      if (va) {
+                        getMessageSignatureMetamask(Web3, sigHash, va).then(
+                          (msg) => {
+                            snapshotCurrentRound(votingContract, msg).then(
+                              (tx) => {
+                                // TODO: Refetch state after snapshot.
+                                console.log("success?", tx);
+                              }
+                            );
+                          }
+                        );
+                      }
+                    }
+                  }
+                );
+              }}
+              variant="secondary"
+            >
+              {signer ? "Snapshot Round" : "Connect Wallet to Snapshot"}
+            </Button>
+          ) : canReveal ? (
+            <Button
+              type="button"
+              onClick={() => {
+                // WIP. Comment out for now.
+                // console.log("encryptedVotes", encryptedVotes);
+                if (encryptedVotes.length && activeRequests.length) {
+                  const postData = [] as PostRevealData[];
+                  activeRequests.forEach((el, index) => {
+                    const datum = {} as PostRevealData;
+                    // I believe latest events are on bottom. requires testing.
+                    const latestVotesFirst = [...encryptedVotes].reverse();
+                    const findVote = latestVotesFirst.find(
+                      (x) => x.identifier === el.identifier
+                    );
+
+                    if (findVote) {
+                      datum.ancillaryData = el.ancillaryData;
+                      // anc data is set to - or N/A in UI if empty, convert back to 0x.
+                      if (
+                        el.ancillaryData === UNDEFINED_VOTE ||
+                        el.ancillaryData === "N/A"
+                      ) {
+                        datum.ancillaryData = "0x";
+                      } else {
+                        datum.ancillaryData = web3.utils.utf8ToHex(
+                          el.ancillaryData
+                        );
+                      }
+                      datum.time = Number(el.time);
+                      datum.identifier = el.idenHex;
+                      datum.salt = findVote.salt;
+                      // datum.price = toWeiSafe(findVote.price).toString();
+                      datum.price = findVote.price.toString();
+                      postData.push(datum);
+                    }
+                  });
+
+                  // Make sure to use the two key contract for revealing if it exists
+                  let vc = votingContract;
+                  if (designatedVotingContract) vc = designatedVotingContract;
+                  if (vc) {
+                    revealVotes(vc, postData).then((res) => {
+                      // refetch votes.
+                      console.log("successfully revealed");
```

Comment:
can prob remove this

Reply:
I actually do some stuff in here (refetch data) in my next branch. I'll just leave it alone.